### PR TITLE
ENH: Implement Stochastic Maps as GATExprs

### PR DESCRIPTION
**Pull request by jpfairbanks**
Rebase of PR closes #157

### CODE REVIEW

File: experiments/Markov/src/StochExpr.jl [hunk 1]
```diff
@@ -0,0 +1,46 @@
+using Catlab
+using Catlab.Doctrines
+import Base: length
+
+export crand
+
+length(::FreeCartesianCategory.Ob{:generator}) = 1
```
Comment (epatters) [inline experiments/Markov/src/StochExpr.jl:7]: This convenient function is also ready implemented in `Catlab.Doctrines` as `ndims`. Maybe it should be `length`? Depends on how you think about the object.

File: experiments/Markov/src/Markov.jl [hunk 2]
```diff
@@ -1,5 +1,7 @@
 module Markov
 
 include("MarkovCategories.jl")
+include("StochMaps.jl")
```
Comment (epatters) [inline experiments/Markov/src/Markov.jl:4]: Should this line be here? I only see `StochExpr.jl`.

**Comment by epatters** [comment]
Maybe a few small comments. Otherwise, LGTM.

File: experiments/Markov/src/Markov.jl (hunk 2 above)
Comment (jpfairbanks) [reply experiments/Markov/src/Markov.jl:4]: Leftover from the last deletion.

**Comment by jpfairbanks** [comment]
🚢 ?

**Comment by epatters** [comment]
🚢-ed!

**Pull request merged**